Va cowamm banner

fairlighteth opened the pull request:
# Summary

- Creates a unified CoW AMM banner component that can serve different variations depending on location of usage. For now `global` and `tokenSelector` are the two available locations.
- DEMO only dropdown activated for this PR
- Includes click analytics events


https://github.com/user-attachments/assets/318aba88-c30d-4565-a009-95426da4a112


https://github.com/user-attachments/assets/61e39640-7800-4a9c-be4d-795a3c77a955

fairlighteth commented:
@elena-zh Did some updates:
- Don't show banners when not connected with a wallet
- Added an extra scenario:

<img width="1153" alt="Screenshot 2024-10-16 at 12 05 49" src="https://github.com/user-attachments/assets/bdd98771-5555-47b5-8b06-eb2f7ac5cb3c">

elena-zh commented:
Hey @fairlighteth , great job!
 
Some cases/issues:
1. Safe or any other SC-wallet: it should not say that transactions are gas-free there
![image](https://github.com/user-attachments/assets/b3845c33-7f37-4261-b153-c0d2ce754c79)
![image](https://github.com/user-attachments/assets/c2debacf-e167-45c6-aa12-40d064a0e6e2)

2. Mobile, real devices: tokens info is not displayed on a banner inside token list: 
![image](https://github.com/user-attachments/assets/189f8a5c-b2a7-43f0-a91f-c762d52abdfe)
3. Colors for banners are differ in the light and dark modes. Is it done intentionally?
![dark](https://github.com/user-attachments/assets/3e920501-4a38-40ea-aa01-5e316bb5bf0b)
![light](https://github.com/user-attachments/assets/6c38f14d-1745-466d-953a-ae5434791c4c)
4. I think, it would be great to add one more case when a user is connected to an unsupported network. IMO, it should behave the same as to a not connected to a wallet?
5. (sorry if I've missed this somewhere else) What should we show when there is no pool in a current network (e.g. Sepolia)? OR no CoW AMM pools in a selected network (again, on Sepolia, but in the future in Base, for example)?

Thanks!

fairlighteth commented:
@elena-zh thank you for reviewing this:
1. Fixed
2. Fixed
3. Done intentionally
4. Yes correct, fixed now
5. Balancer supports these networks, which we all cover (except for Sepolia, but that's just for testing anyway...)
<img width="355" alt="Screenshot 2024-10-18 at 14 24 25" src="https://github.com/user-attachments/assets/8fe8c96f-abb4-419e-93a3-69a3959d588e">

So in reference to point 4, right now we only show the banners for networks we support in our UI.

elena-zh commented:
@fairlighteth , thank you!

To clarify the 5th case I reported above: 
We don't have CoW AMM pools on Sepolia on Prod (however, it is a supported network). 
Same as for other sources. 
My question was more about what message should we show on the c=CoW AMM banner in this case?

fairlighteth commented:
@elena-zh for now I think we show the general message, which would link you to CoW AMM balancer's page. For now I wouldn't make the message to specific saying "Switch to Ethereum and explore pools" etc.

elena-zh reviewed (approved):
Approved then! :)

alfetopito commented on apps/cowswap-frontend/src/common/pure/CoWAMMBanner/CoWAmmBannerContent.tsx:
```diff
@@ -0,0 +1,396 @@
+  onCtaClick,
+  onClose,
+}: CoWAmmBannerContentProps) {
+  const isMobile = useMediaQuery(upToSmall)
+  const isDarkMode = useIsDarkMode()
```
Ideally these should come as props, given this is a pure component

shoom3301 replied:
I will refactor that part along with connecting it to the logic

alfetopito commented on apps/cowswap-frontend/src/modules/tokensList/pure/TokensVirtualList/index.tsx:
```diff
@@ -33,7 +38,11 @@ export function TokensVirtualList(props: TokensVirtualListProps) {
     props
   const { values: balances, isLoading: balancesLoading } = balancesState
 
+  const { account: connectedAccount } = useWalletInfo()
+
+  const isInjectedWidgetMode = isInjectedWidget()
   const isWalletConnected = !!account
+  const isChainIdUnsupported = useIsProviderNetworkUnsupported()
```
Same thing, pure component, don't use hooks

shoom3301 replied:
I will refactor that part along with connecting it to the logic
